src/AnalyzerPowerPack/Core/Design/CA1052DiagnosticAnalyzer.cs

```diff
@@ -100,13 +104,18 @@ internal static class CA1052Extensions
         /// "qualifying member" (<see cref="IsQualifyingMember(ISymbol)"/>) and no
         /// "disqualifying members" (<see cref="IsDisqualifyingMember(ISymbol)"/>).
         /// </remarks>
-        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol)
+        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol, Compilation compilation)
         {
             if (symbol.TypeKind != TypeKind.Class)
             {
                 return false;
             }
 
+            if (symbol.BaseType != compilation.GetTypeByMetadataName("System.Object"))
```

Comment:
>You can simply say symbol.BaseType.SpecialType != SpecialType.System_Object and avoid passing in the compilation parameter.

Thanks, fixed.

>And you probably want to check for existence of BaseType so symbol.BaseType?.SpecialType != SpecialType.System_Object

Can you suggest a test case with BaseType==null at this point? We have already ensured that this is a class and that it is not static.